Tests/Observable+MoyaSpec.swift

```diff
@@ -266,5 +266,103 @@ class ObservableMoyaSpec: QuickSpec {
+                expect(receivedObject?.createdAt) == formatter.date(from: "1995-01-14T12:34:56")!
+            }
+
+            it("maps data representing a json array to an array of decodable objects") {
+                let jsonArray = [json, json, json]
+                guard let data = try? JSONSerialization.data(withJSONObject: jsonArray, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var receivedObjects: [Issue]?
+                _ = observable.map([Issue].self, using: decoder).subscribe(onNext: { objects in
+                    receivedObjects = objects
+                })
+                expect(receivedObjects).notTo(beNil())
+                expect(receivedObjects?.count) == 3
+                expect(receivedObjects?.map { $0.title }) == ["Hello, Moya!", "Hello, Moya!", "Hello, Moya!"]
+            }
+
+            it("maps data representing a json at a key path to a decodable object") {
+                let json: [String: Any] = ["issue": json] // nested json
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var receivedObject: Issue?
+                _ = observable.map(Issue.self, atKeyPath: "issue", using: decoder).subscribe(onNext: { object in
+                    receivedObject = object
+                })
+                expect(receivedObject).notTo(beNil())
+                expect(receivedObject?.title) == "Hello, Moya!"
+                expect(receivedObject?.createdAt) == formatter.date(from: "1995-01-14T12:34:56")!
+            }
+
+            it("ignores invalid data") {
+                var json = json
+                json["createdAt"] = "Hahaha" // invalid date string
+                guard let data = try? JSONSerialization.data(withJSONObject: json, options: .prettyPrinted) else {
+                    preconditionFailure("Failed creating Data from JSON dictionary")
+                }
+                let observable = Response(statusCode: 200, data: data).asObservable()
+
+                var receivedError: Error?
+                _ = observable.map(Issue.self, using: decoder).subscribe { event in
+                    switch event {
+                    case .next:
+                        fail("next called for invalid data")
+                    case .error(let error):
+                        receivedError = error
+                    default:
+                        break
+                    }
+                }
+
+                // check whether we properly got a MoyaError
+                expect(receivedError).to(Predicate<Error> { expression in
```

Comment:
hmm, isn't `MoyaError` equatable? if not, maybe we can make it so (at least for tests), would make this test cleaner

Reply:
It would be tough because `MoyaError.objectMapping` has an associated value `Error` which is not an Equatable. I ended up with using `if-case-let` instead.

```swift
if case let MoyaError.objectMapping(nestedError, _)? = receivedError {
    expect(nestedError).to(beAKindOf(DecodingError.self))
} else {
    fail("expected <MoyaError.objectMapping>, got <\(String(describing: receivedError))>")
}
```